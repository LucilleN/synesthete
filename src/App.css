.App {
  /* margin-left: 5rem;
  margin-right: 5rem; */
  width: 100%;
  /* padding-left: 5rem;
  padding-right: 5rem; */
}
