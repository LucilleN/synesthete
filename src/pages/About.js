import React from 'react'

import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import StartButton from '../components/StartButton'

const styles = theme => ({
  root: {
    textAlign: 'center',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center',
    overflow: 'auto',
    maxHeight: 'calc(100% - 60px)',
    minHeight: 'calc(100% - 60px)',
    paddingBottom: 10
  },
  title: {
    color: theme.palette.light.pink,
    // padding: 5,
    fontSize: '3rem',
    textAlign: 'center'
  },
  section: {
    textAlign: 'center',
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    marginTop: 10,
    marginBottom: 10
  },
  sectionHeading: {
    fontSize: '2rem',
    fontWeight: 600,
    color: theme.palette.white,
    opacity: '0.6',
    marginBottom: 5
  },
  paragraph: {
    fontSize: '1rem',
    lineHeight: '1 rem',
    color: theme.palette.white,
    opacity: '0.4',
    textAlign: 'center',
    width: '60%',
    marginBottom: 10
  }
})

const About = props => {
  const { classes } = props
  return (
    <div className={classes.root}>
      <Typography className={classes.title}>
        about
      </Typography>
      <div className={classes.section}>
        <Typography className={classes.sectionHeading}>
          synesthesia, chromesthesia, and synesthete
        </Typography>
        <Typography className={classes.paragraph}>
          Synesthesia is a neurological phenomenon that allows the senses to overlap.
          Roughly 4% of people have some form of synesthesia, and others have experienced it under the influence of psychedelic drugs.
          Chromesthetes in particular have sound-color synesthesia, which means that they hear music as colors.
          This makes listening to music an especially vibrant experience.
        </Typography>
        <Typography className={classes.paragraph}>
          Many famous people, both past and present, have had this phenomenon, including Pythagoras, Isaac Newton, George Field, Alexander Scriabin, Duke Ellington, Pharrell Williams, Stevie Wonder, and Kanye West.
          Synesthesia is, unsurprisingly, up to 7 times more common in artists and musicians, and many of them use their synesthesia in developing their artwork.
          However, synesthesia is different for everyone.
          Synesthetes can have any crossover of the senses -- sight, sound, smell, taste, and touch -- and can involve more than just two.
        </Typography>
        <Typography className={classes.paragraph}>
          When it comes to chromesthesia (sounds to colors), people's perceptions of the colors of music vary, too.
          Many chromesthetes see the same fixed hue for each note -- red for C and blue for G, for example.
          For others, the color spectrum is transposed.
        </Typography>
        <Typography className={classes.paragraph}>
          In Synesthete, each song has the full color spectrum, but the rainbow is offset based on the song's key.
          Other aspects of the song, such as its tempo, valence, and acousticness, affect other aspects of the visualization.
          It's not possible to simulate exactly what chromesthesia is like, but this is our closest bet -- so grab a pair of headphones and start hearing music in color!
        </Typography>
      </div>
      <div className={classes.section}>
        <Typography className={classes.sectionHeading}>
          the creator
        </Typography>
        <Typography className={classes.paragraph}>
          Synesthete was created by Lucille Njoo for CMSI 370.
        </Typography>
      </div>
    </div>
  )
}

export default withStyles(styles)(About)
