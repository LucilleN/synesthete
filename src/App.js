import React, { useEffect } from 'react'
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'

import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import { withStyles } from '@material-ui/core/styles'

import Homepage from './pages/Homepage'
import About from './pages/About'
import Search from './pages/Search'
import UploadVisualization from './pages/UploadVisualization'
import Visualization from './pages/Visualization'
import MenuBar from './components/MenuBar'

import { apiHost } from './api'

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#c23063', // magenta
      dark: '#180142' // dark purple
    },
    type: 'dark',
    dark: {
      pink: '#bf265c',
      purple: '#1c004f',
    },
    medium: {
      purple: '#5a0b66',
    },
    light: {
      pink: '#db91aa',
      purple: '#ac91db'
    },
    white: '#FFFFFF',
  },
  typography: {
    fontFamily: ['Dosis', 'Roboto', 'Open Sans', '"Helvetica Neue"', 'Arial', 'sans-serif'].join(',')
  }
})

export const defaultErrorText = 'Sorry, but something went wrong.'

const styles = () => ({
  root: {
    width: '100vw',
    height: '100%',
    minHeight: '100vh',
    background: `linear-gradient(0deg, ${theme.palette.dark.pink} 0%, ${theme.palette.dark.purple} 80%)`
  }
})

const routes = (
  <Router>
    <div>
      <MenuBar />
      <Switch>
        <Route exact path="/" component={Homepage} />
        <Route exact path="/about" component={About} />
        <Route exact path="/search" component={Search} />
        <Route exact path="/upload" component={UploadVisualization} />
        <Route exact path="/visualization" component={Visualization} />
        <Route exact path="/visualization/:songID" component={Visualization} />
      </Switch>
    </div>
  </Router>
)

const App = (props) => {

  useEffect(() => apiHost('https://api.spotify.com/v1/'))

  const { classes } = props

  return (
    <div className={classes.root}>
      <MuiThemeProvider theme={theme}>
        {routes}
      </MuiThemeProvider>
    </div>
  )
}

export default withStyles(styles)(App)
