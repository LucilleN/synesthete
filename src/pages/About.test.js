import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'
import { MuiThemeProvider } from '@material-ui/core/styles'

// import sinon from 'sinon'

import { theme } from '../App'
import About from './About'
// import * as api from '../api'

let component;
beforeEach(() => {
  component = TestRenderer.create(
    <MuiThemeProvider theme={theme}>
      <About />
    </MuiThemeProvider>
  )
})

it('should have a title', () => {
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should have two text sections', () => {
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should have a sectionHeading for each section', () => {
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


// describe('search button', () => {
//   let div
//   beforeEach(() => {
//     div = document.createElement('div')
//     ReactTestUtils.act(() => {
//       ReactDOM.render(
//         <MuiThemeProvider theme={theme}>
//           <Search />
//         </MuiThemeProvider>
//       , div)
//     })
//   })
//
//   afterEach(() => ReactDOM.unmountComponentAtNode(div))
//
//   it('should be enabled when the search field is not blank', () => {
//     const searchInput = div.querySelector('input')
//     ReactTestUtils.act(() => {
//       searchInput.value = 'i can haz unit tests'
//       ReactTestUtils.Simulate.change(searchInput)
//     })
//
//     const searchButton = div.querySelector('button')
//     expect(searchButton.disabled).toBe(false)
//   })
//
//   it('should be disabled when the query is blank', () => {
//     const searchButton = div.querySelector('button')
//     expect(searchButton.disabled).toBe(true)
//   })
//
//   it('should be enabled when the user enters a query', () => {
//     const searchInput = div.querySelector('input')
//     ReactTestUtils.act(() => {
//       searchInput.value = 'hello there this is a song search'
//       ReactTestUtils.Simulate.change(searchInput)
//     })
//
//     const searchButton = div.querySelector('button')
//     expect(searchButton.disabled).toBe(false)
//   })
// })
