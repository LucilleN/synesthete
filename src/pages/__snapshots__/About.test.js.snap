// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have a sectionHeading for each section 1`] = `
<div
  className="About-root-1"
>
  <p
    className="MuiTypography-root About-title-2 MuiTypography-body1"
  >
    about
  </p>
  <div
    className="About-section-3"
  >
    <p
      className="MuiTypography-root About-sectionHeading-4 MuiTypography-body1"
    >
      synesthesia, chromesthesia, and synesthete
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Synesthesia is a neurological phenomenon that allows the senses to overlap. Roughly 4% of people have some form of synesthesia, and others have experienced it under the influence of psychedelic drugs. Chromesthetes in particular have sound-color synesthesia, which means that they hear music as colors. This makes listening to music an especially vibrant experience.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Many famous people, both past and present, have had this phenomenon, including Pythagoras, Isaac Newton, George Field, Alexander Scriabin, Duke Ellington, Pharrell Williams, Stevie Wonder, and Kanye West. Synesthesia is, unsurprisingly, up to 7 times more common in artists and musicians, and many of them use their synesthesia in developing their artwork. However, synesthesia is different for everyone. Synesthetes can have any crossover of the senses -- sight, sound, smell, taste, and touch -- and can involve more than just two.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      When it comes to chromesthesia (sounds to colors), people's perceptions of the colors of music vary, too. Many chromesthetes see the same fixed hue for each note -- red for C and blue for G, for example. For others, the color spectrum is transposed.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      In Synesthete, each song has the full color spectrum, but the rainbow is offset based on the song's key. Other aspects of the song, such as its tempo, valence, and acousticness, affect other aspects of the visualization. It's not possible to simulate exactly what chromesthesia is like, but this is our closest bet -- so grab a pair of headphones and start hearing music in color!
    </p>
  </div>
  <div
    className="About-section-3"
  >
    <p
      className="MuiTypography-root About-sectionHeading-4 MuiTypography-body1"
    >
      the creator
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Synesthete was created by Lucille Njoo for CMSI 370.
    </p>
  </div>
</div>
`;

exports[`should have a title 1`] = `
<div
  className="About-root-1"
>
  <p
    className="MuiTypography-root About-title-2 MuiTypography-body1"
  >
    about
  </p>
  <div
    className="About-section-3"
  >
    <p
      className="MuiTypography-root About-sectionHeading-4 MuiTypography-body1"
    >
      synesthesia, chromesthesia, and synesthete
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Synesthesia is a neurological phenomenon that allows the senses to overlap. Roughly 4% of people have some form of synesthesia, and others have experienced it under the influence of psychedelic drugs. Chromesthetes in particular have sound-color synesthesia, which means that they hear music as colors. This makes listening to music an especially vibrant experience.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Many famous people, both past and present, have had this phenomenon, including Pythagoras, Isaac Newton, George Field, Alexander Scriabin, Duke Ellington, Pharrell Williams, Stevie Wonder, and Kanye West. Synesthesia is, unsurprisingly, up to 7 times more common in artists and musicians, and many of them use their synesthesia in developing their artwork. However, synesthesia is different for everyone. Synesthetes can have any crossover of the senses -- sight, sound, smell, taste, and touch -- and can involve more than just two.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      When it comes to chromesthesia (sounds to colors), people's perceptions of the colors of music vary, too. Many chromesthetes see the same fixed hue for each note -- red for C and blue for G, for example. For others, the color spectrum is transposed.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      In Synesthete, each song has the full color spectrum, but the rainbow is offset based on the song's key. Other aspects of the song, such as its tempo, valence, and acousticness, affect other aspects of the visualization. It's not possible to simulate exactly what chromesthesia is like, but this is our closest bet -- so grab a pair of headphones and start hearing music in color!
    </p>
  </div>
  <div
    className="About-section-3"
  >
    <p
      className="MuiTypography-root About-sectionHeading-4 MuiTypography-body1"
    >
      the creator
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Synesthete was created by Lucille Njoo for CMSI 370.
    </p>
  </div>
</div>
`;

exports[`should have two text sections 1`] = `
<div
  className="About-root-1"
>
  <p
    className="MuiTypography-root About-title-2 MuiTypography-body1"
  >
    about
  </p>
  <div
    className="About-section-3"
  >
    <p
      className="MuiTypography-root About-sectionHeading-4 MuiTypography-body1"
    >
      synesthesia, chromesthesia, and synesthete
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Synesthesia is a neurological phenomenon that allows the senses to overlap. Roughly 4% of people have some form of synesthesia, and others have experienced it under the influence of psychedelic drugs. Chromesthetes in particular have sound-color synesthesia, which means that they hear music as colors. This makes listening to music an especially vibrant experience.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Many famous people, both past and present, have had this phenomenon, including Pythagoras, Isaac Newton, George Field, Alexander Scriabin, Duke Ellington, Pharrell Williams, Stevie Wonder, and Kanye West. Synesthesia is, unsurprisingly, up to 7 times more common in artists and musicians, and many of them use their synesthesia in developing their artwork. However, synesthesia is different for everyone. Synesthetes can have any crossover of the senses -- sight, sound, smell, taste, and touch -- and can involve more than just two.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      When it comes to chromesthesia (sounds to colors), people's perceptions of the colors of music vary, too. Many chromesthetes see the same fixed hue for each note -- red for C and blue for G, for example. For others, the color spectrum is transposed.
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      In Synesthete, each song has the full color spectrum, but the rainbow is offset based on the song's key. Other aspects of the song, such as its tempo, valence, and acousticness, affect other aspects of the visualization. It's not possible to simulate exactly what chromesthesia is like, but this is our closest bet -- so grab a pair of headphones and start hearing music in color!
    </p>
  </div>
  <div
    className="About-section-3"
  >
    <p
      className="MuiTypography-root About-sectionHeading-4 MuiTypography-body1"
    >
      the creator
    </p>
    <p
      className="MuiTypography-root About-paragraph-5 MuiTypography-body1"
    >
      Synesthete was created by Lucille Njoo for CMSI 370.
    </p>
  </div>
</div>
`;
